x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/bucket_nesting_editor.test.tsx

```diff
@@ -0,0 +1,215 @@
+ */
+
+import { mount } from 'enzyme';
+import React from 'react';
+import { BucketNestingEditor } from './bucket_nesting_editor';
+import { IndexPatternColumn } from '../indexpattern';
+
+describe('BucketNestingEditor', () => {
+  function mockCol(col: Partial<IndexPatternColumn> = {}): IndexPatternColumn {
+    const result = {
+      dataType: 'string',
+      isBucketed: true,
+      label: 'a',
+      operationType: 'terms',
+      params: {
+        size: 5,
+        orderBy: { type: 'alphabetical' },
+        orderDirection: 'asc',
+      },
+      sourceField: 'a',
+      suggestedPriority: 0,
+      ...col,
+    };
+
+    return result as IndexPatternColumn;
+  }
+
+  it('should display a checked switch if there are two buckets and it is the root', () => {
+    const component = mount(
+      <BucketNestingEditor
+        columnId="a"
+        layer={{
+          columnOrder: ['a', 'b', 'c'],
+          columns: {
+            a: mockCol({ suggestedPriority: 0 }),
+            b: mockCol({ suggestedPriority: 1 }),
+            c: mockCol({ suggestedPriority: 2, operationType: 'min', isBucketed: false }),
+          },
+          indexPatternId: 'foo',
+        }}
+        setColumns={jest.fn()}
+      />
+    );
+    const control = component.find('[data-test-subj="indexPattern-nesting-switch"]').first();
+
+    expect(control.prop('checked')).toBeFalsy();
+  });
+
+  it('should display an unchecked switch if there are two buckets and it is not the root', () => {
+    const component = mount(
+      <BucketNestingEditor
+        columnId="a"
+        layer={{
+          columnOrder: ['b', 'a', 'c'],
+          columns: {
+            a: mockCol({ suggestedPriority: 0 }),
+            b: mockCol({ suggestedPriority: 1 }),
+            c: mockCol({ suggestedPriority: 2, operationType: 'min', isBucketed: false }),
+          },
+          indexPatternId: 'foo',
+        }}
+        setColumns={jest.fn()}
+      />
+    );
+    const control = component.find('[data-test-subj="indexPattern-nesting-switch"]').first();
+
+    expect(control.prop('checked')).toBeTruthy();
+  });
+
+  it('should reorder the columns when toggled', () => {
+    const setColumns = jest.fn();
+    const component = mount(
+      <BucketNestingEditor
+        columnId="a"
+        layer={{
+          columnOrder: ['b', 'a', 'c'],
+          columns: {
+            a: mockCol({ suggestedPriority: 0 }),
+            b: mockCol({ suggestedPriority: 1 }),
+            c: mockCol({ suggestedPriority: 2, operationType: 'min', isBucketed: false }),
+          },
+          indexPatternId: 'foo',
+        }}
+        setColumns={setColumns}
+      />
+    );
+    const control = component.find('[data-test-subj="indexPattern-nesting-switch"]').first();
+
+    (control.prop('onChange') as () => {})();
+
+    expect(setColumns).toHaveBeenCalledWith(['a', 'b', 'c']);
+  });
+
+  it('should display nothing if there are no buckets', () => {
```

Comment:
Seems like this is the same test as the one above, maybe you forgot to implement it?

Reply:
Whoops. You're right. Copy/paste for the win.